### PR TITLE
Fix upload and wrote tests

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This is only in dev/master, so no backport needed.
:+1:  for using dev myself and suffering from it ;-)
 
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

File: src/androidTest/java/com/owncloud/android/UploadIT.java [hunk 1]
```diff
@@ -183,4 +184,151 @@ public void testUploadInNonExistingFolder() {
...
+        newUpload.setRemoteFolderToBeCreated();
+        newUpload.addRenameUploadListener(() -> {
+            // dummy
+        });
+
+        RemoteOperationResult result = newUpload.execute(client, getStorageManager());
+        assertTrue(result.toString(), result.isSuccess());
+
+        // cleanup
+        new RemoveFileOperation("/testUpload/", false, account, false, targetContext).execute(client, getStorageManager());
```
Comment (ezaquarii) [inline src/androidTest/java/com/owncloud/android/UploadIT.java:262]: what if that fails? can the test be repeated again?

File: src/androidTest/java/com/owncloud/android/UploadIT.java [hunk 2]
```diff
@@ -183,4 +184,151 @@ public void testUploadInNonExistingFolder() {
...
+            public BatteryStatus getBattery() {
+                return new BatteryStatus(true, 100);
+            }
+        };
+
+        OCUpload ocUpload = new OCUpload(FileStorageUtils.getSavePath(account.name) + "/empty.txt",
+                                         "/testUpload/charging.txt", account.name);
+        ocUpload.setWhileChargingOnly(true);
+
+        UploadFileOperation newUpload = new UploadFileOperation(
```
Comment (ezaquarii) [inline src/androidTest/java/com/owncloud/android/UploadIT.java:240]: This setup seems to be repetive. We could do it in:

```
@Before
void setUp() { ... }
```

**Comment by ezaquarii** [comment]
```
com.owncloud.android.util.EncryptionTestIT > testCryptFileWithoutMetadata[android-27(AVD) - 8.1.0] FAILED
--
(...) Instrumentation process crashed.
```

File: src/androidTest/java/com/owncloud/android/UploadIT.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/androidTest/java/com/owncloud/android/UploadIT.java:262]: If this fails something is seriously broken and then I think it is ok that this (or any following) test fails.

File: src/androidTest/java/com/owncloud/android/UploadIT.java (hunk 2 above)
Comment (tobiasKaminsky) [reply src/androidTest/java/com/owncloud/android/UploadIT.java:240]: It is, and for unmodified uploads I have "testUpload(…)".
But the 4 new added tests have slightly modifications:
- use special connectivtyServiceMock / powerManagementServiceMock
- use different values for onWifiOnly/whileChargingOnly

So for better readability I would keep it this way.

**Pull request merged**